keras_core/distribute/distribution.py

```diff
@@ -0,0 +1,144 @@
+"""Unified high level distribution APIs across backends.
+
+!!!DO NOT USE!!! Currently under development and APIs are not final.
+
+Currently only the JAX backend has been implemented, and the Tensorflow backend
+will be implemented in future (via tf.dtensor API).
+"""
+
+import contextlib
+from typing import Any
+
+from keras_core import KerasVariable
+from keras_core.backend.common import global_state
+
+GLOBAL_ATTRIBUTE_NAME = "distribution"
+
+
+def list_devices(device_type: str = None) -> list[str]:
+    """Return all the available devices based on the device type.
+
+    Note that this should return the global devices in a distributed setting.
+
+    Args:
+        device_type: CPU, GPU or TPU. Default to GPU or TPU if available when
+            device_type is not provided. Otherwise will return the CPU devices.
+
+    Return:
+        List of devices that are available for distribute computation.
+    """
+    pass
+
+
+class DeviceMesh:
+    """The cluster of computation devices for distributed computation.
+
+    This is aligned with `jax.sharding.Mesh` and `tf.dtensor.Mesh`, which
+    represents the computation devices in the global context.
+
+    See more details in
+    https://jax.readthedocs.io/en/latest/jax.sharding.html#jax.sharding.Mesh
+    and https://www.tensorflow.org/api_docs/python/tf/experimental/dtensor/Mesh.
+    """
+
+    def __init__(
+        self,
+        shape: list | tuple,
+        axis_names: list[str],
+        devices: list[str] = None,
+    ):
+        """Initialize the DeviceMesh for the given topology.
+
+        Args:
+            shape: the shape of the overall DeviceMesh, e.g. (8,) for a data
+                parallel only distribution, or (4, 2) for a model+data parallel
+                distribution.
+            axis_names: The logical name of the each axis for the DeviceMesh.
+                The length of the `axis_names` should match to the rank of the
+                `shape`. The `axis_names` will be used to match/create the
+                `TensorLayout` when distribute the data and weights.
+            devices: Optional list of devices. Default to all the available
+                devices locally from `list_devices()`.
+        """
```

Comment:
It would be useful to specify a couple of code examples for how to create a `DeviceMesh` and a `TensorLayout`

Reply:
Ack, once I add some unit test after the real code change, I will put them here as the example.